kolibri/core/assets/src/core_app_constructor.js

```diff
@@ -44,15 +48,22 @@ function Lib() {
  */
 module.exports = function CoreApp() {
   this.lib = new Lib();
+  this.resources = new ResourceManager(this);
   const mediator = new Mediator();
 
+  this.resources.registerResource('contentmetadata', 'id', ContentMetaDataResource);
```

Comment:
Oh, I didn't understand that it's in fact the resource manager doing the constructing.

Tying in with the comment above though, this could potentially just read:

```javascript
this.resources.registerResource(ContentMetaDataResource);
```

If resources define their own URL name and key ID.